sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -49,16 +49,16 @@ class Iso8601TimestampFormatter(
   @transient
   protected lazy val formatter = getOrCreateFormatter(pattern, locale)
 
-  private def toInstant(s: String): Instant = {
-    val temporalAccessor = formatter.parse(s)
-    if (temporalAccessor.query(TemporalQueries.offset()) == null) {
-      toInstantWithZoneId(temporalAccessor, timeZone.toZoneId)
-    } else {
-      Instant.from(temporalAccessor)
-    }
-  }
+  override def parse(s: String): Long = {
+    val parsed = formatter.parse(s)
+    val parsedZoneId = parsed.query(TemporalQueries.zone())
+    val zoneId = if (parsedZoneId == null) timeZone.toZoneId else parsedZoneId
```

Comment:
Do we need to evaluate `timeZone.toZoneId` everytime? Can we evalute this once and store this into `val` when this class is instantiated?

Reply:
> Do we need to evaluate timeZone.toZoneId everytime?

No, we don't.

> Can we evaluate this once and store this into val when this class is instantiated?

Sure. Actually I think of go over all time-related code in a separate PR, and try to replace `TimeZone` objects by `ZoneId` or even by `ZoneOffset`.